packages/smart-wallet/src/smartWallet.js

```diff
@@ -363,6 +364,28 @@ export const prepareSmartWallet = (baggage, shared) => {
 
   const makeAmountWatcher = prepareAmountWatcher();
 
+  /**
+   * @param {Amount} amount
+   * @returns {Amount}
+   */
+  const cleanAmountForPublish = amount => {
+    mustMatch(amount, AmountShape);
+    if (
+      !matches(amount.brand, shared.invitationBrand) ||
+      !matches(amount.value, harden([InvitationElementShape]))
```

Comment:
Can you please explain why this code assumes there is only a single invitation in the invitation amount?  I don't think that's generally true, and may be fragile.

If it is true, please write a comment explaining why.

The following match bails without filtering if there is more than one invitation.  Maybe instead try:
```suggestion
      !matches(amount.value, M.arrayOf(InvitationElementShape))
```